spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -520,6 +520,20 @@ class DeltaCatalog extends DelegatingCatalogExtension
 
   /** Checks if a table already exists for the provided identifier. */
   def getExistingTableIfExists(table: TableIdentifier): Option[CatalogTable] = {
+    try {
+      val loadedTable =
+        super.loadTable(Identifier.of(Array(table.database.getOrElse("")), table.table))
```

Comment:
shall we check table existence before loading the table? then we can save the try-catch and it's consistent with `getExistingTableIfExistsFromHMS`